apps/spark/src/main/java/com/linkedin/openhouse/jobs/scheduler/tasks/OperationTask.java

```diff
@@ -142,12 +142,13 @@ private Optional<JobState> pollJobStatus(Attributes typeAttributes) {
     long startTime = System.currentTimeMillis();
     while (!jobFinished()) {
       long elapsedTime = System.currentTimeMillis() - startTime;
-      if (elapsedTime > timeoutMs) {
+      // Cancel job if it is queued for more than queuedTimeoutMs.
+      // Otherwise, if it is running, then keep it running until scheduler cancels it.
+      if (elapsedTime > queuedTimeoutMs) {
         Optional<JobState> job = jobsClient.getState(jobId);
-        // Do not cancel job if it is still in running state after 3 hours
-        if (job.isPresent() && !job.get().equals(JobState.RUNNING)) {
```

Comment:
On the other hand job can run for longer duration like OFD or compaction. Compaction can take more than an hour. Should we make the running job timeout configurable? For snapshot expiration we can set very less timeout. For compaction we can keep 30 min or 1 hour. If configurable we can set different timeout for different job types.  Post the timeout if the job still in running state we can just exit without cancelling the job. So we should have timeout for other job state as well not just queued ones.

Reply:
We never cancelled the running jobs, unless the scheduler reaches 12 hour SLA. I didn't change the logic here, but just shorten the wait time for **QUEUED** jobs. So there's no need for additional configuration.